SA3/qa_web/views.py

```diff
@@ -29,6 +29,41 @@ def index(request):
     return render(request, 'qa_web/index.html', context={'questions': Questions.objects.all()})
 
 
+@login_required(login_url='/login/')
+def edit_profile(request):
+    if request.method=='GET':
+        form = UserProfile()
+        # return render_to_response('qa_web/UserProfile.html', RequestContext(request, {'form': form, }))
+        return render(request,'qa_web/EditUserProfile.html', context={'form': form})
+    else:
+        form = UserProfile(request.POST)
+        # validating the form occurs below and then saving the results entered by the user
+        if form.is_valid():
+            user = request.user
+
+            user.first_name = request.POST['prename']
+            user.last_name = request.POST['surname']
+            user.age = request.POST['age']
+            user.birthday = request.POST['birthday']
+            user.motherland = request.POST['motherland']
+            user.school= request.POST['school']
+            user.major = request.POST['major']
+            user.city = request.POST['city']
+
+            user.save()
+
+            # when the information is entered and the information is saved
+            # the page gets redirected to the profile page
+            return HttpResponseRedirect('/profile/' + str(user.id))
+        else:
+            return render(request, 'qa_web/EditUserProfile.html', context={'form': form})
```

Comment:
Great use of form server-side validation!

Reply:
why thank you.